test/functionalTest/cases/2937_self_notification_protection/self_notification_one_hop.test

```diff
@@ -0,0 +1,193 @@
+payload='{
+  "id": "E",
+  "type": "Thing",
+  "A": {
+    "value": "boom",
+    "type": "Text"
+  }
+}'
+orionCurl --url /v2/entities --payload "$payload"
+echo
+echo
+
+
+echo "03. Wait for a while and check that only one notification was sent"
+echo "=================================================================="
+sleep 5s
+orionCurl --url /v2/subscriptions
+echo
+echo
+
+
+echo "04. Get entity with value boomx"
+echo "==============================="
+orionCurl --url /v2/entities/E
+echo
+echo
+
+
+echo "05. Check the warning about loop detection in the log"
+echo "====================================================="
+egrep 'WARN' /tmp/contextBroker.log
+echo
+echo
+
+
+--REGEXPECT--
+01. Create self-subscription for entity E attribute A
+=====================================================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/subscriptions/REGEX([0-9a-f]{24})
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+02. Create entity with A = boom
+===============================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/entities/E?type=Thing
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+03. Wait for a while and check that only one notification was sent
+==================================================================
+HTTP/1.1 200 OK
+Content-Length: 431
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+[
+    {
+        "id": "REGEX([0-9a-f]{24})",
+        "notification": {
+            "attrs": [],
+            "attrsFormat": "normalized",
+            "httpCustom": {
+                "headers": {
+                    "Content-type": "application/json"
+                },
+                "payload": "%7B %22A%22: %7B %22value%22: %22${A}x%22 %2C %22type%22: %22Text%22 %7D %7D",
+                "url": "http://localhost:REGEX(\d+)/v2/entities/E/attrs"
+            },
+            "lastNotification": "REGEX(.*)",
+            "timesSent": 1
+        },
+        "status": "active",
+        "subject": {
+            "condition": {
+                "attrs": [
+                    "A"
+                ]
+            },
+            "entities": [
+                {
+                    "id": "E"
+                }
+            ]
+        }
+    }
+]
+
+
+04. Get entity with value boomx
+===============================
+HTTP/1.1 200 OK
+Content-Length: 75
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{
+    "A": {
+        "metadata": {},
+        "type": "Text",
+        "value": "boomx"
+    },
+    "id": "E",
+    "type": "Thing"
+}
+
+
+05. Check the warning about loop detection in the log
+=====================================================
+REGEX(.*) msg=Notification loop detected for entity id <E> type <Thing>, skipping subscription triggering
+REGEX(.*) msg=Raising alarm BadInput : intent to update error status of non-existing subscription
```

Comment:
In this case, I prefer not "hiding" warnings. I mean, the right execution means only warning due to loop protection. Any other warning is potentially due to a bug, so I prefer raising it as a fail in the .test execution.

In this case, what we should do is to find out what the wrong warning takes place before merging this PR (and change the .test to remove it, once it gets fixed, either in this PR or in a next one depending of the nature and complexity of the bug).